src/script.js

```diff
@@ -1,18 +1,121 @@
 let textarea;
 let keyboard;
 
-function imitateMouseDown(keyElement) {
+function onKeyDown(keyElement) {
   if (keyElement !== undefined) {
     keyElement.dispatchEvent(new Event('mousedown'));
   }
 }
 
-function imitateMouseUp(keyElement) {
+function onSpecialKeyDown(keyContent) {
+  if (keyContent !== undefined) {
+    const keys = [...keyboard.elements.keys];
+    const keyElement = keys.find((element) => element.textContent === keyContent);
+    keyElement.dispatchEvent(new Event('mousedown'));
+  }
+}
+
+function onKeyUp(keyElement) {
   if (keyElement !== undefined) {
     keyElement.dispatchEvent(new Event('mouseup'));
   }
 }
 
+function onSpecialKeyUp(keyContent) {
+  if (keyContent !== undefined) {
+    const keys = [...keyboard.elements.keys];
+    const keyElement = keys.find((element) => element.textContent === keyContent);
+    keyElement.dispatchEvent(new Event('mouseup'));
+  }
+}
+
+function handleAltShift() {
+  let { locale } = keyboard.properties;
+  if (locale === 'rus') {
+    locale = 'en';
+  } else {
+    locale = 'rus';
+  }
+  window.localStorage.setItem('keyboardLocale', locale);
+  document.body.removeChild(keyboard.elements.main);
+  keyboard.init();
+}
+
+function handleBackspaceClick() {
+  const index = textarea.getCaret();
+  let value = textarea.getValue();
```

Comment:
if add those `index` and `value` as parameter, there will be no need to get it every time. Same for others